### ISSUE TITLE
Paginate rpc commands that return long lists

### ISSUE THREAD

**Issue by mb300sd** [ElementsProject/lightning#4687]
Mentioned briefly in #4305, now coming up again over on RTL ( https://github.com/Ride-The-Lightning/RTL/issues/737 )

lnd has an index/count option, something similar would be very nice to have on c-lightning.

**Comment by whitslack** [comment]
Concept NACK.

Pagination against a live data source is inherently racy. If you need to paginate a large JSON array, capture the entire array into a file and use a JSON manipulation tool (I highly recommend [jq](https://stedolan.github.io/jq/)) to select subranges from it as you desire.

**Comment by mb300sd** [comment]
Maybe a before x/after y timestamp option would be better then, basically anything to avoid having to return millions of results every time listforwards is called.

**Comment by whitslack** [comment]
@mb300sd: Hint: you can also just query the database directly. And if you're using PostgreSQL as your backend, you can add a trigger that notifies you of any insertions or updates to the `forwarded_payments` table as they happen.

**Comment by zerofeerouting** [comment]
I would really like to emphasize the importance of this. Especially with listforwards we're talking hundreds of thousands, if not millions of entries. Would also be great being able to filter results doing `start_date=-2d` or `end_date=-12h`.

**Comment by cdecker** [comment]
I think @rustyrussell was going to design a generic pagination system that could leverage the DB that is backing most of the queries to limit results, order by field and offset based on the ordering column. Maybe he can share the design in a quick doc and maybe we can get this unblocked a bit :-)

**Comment by sangbida** [comment]
This issue is being closed as part of issue cleanup.

**Reason**: About deprecated feature

If you believe this issue should remain open, please comment and we can reopen it.
